standalone-metastore/metastore-catalog/src/main/java/org/apache/iceberg/rest/HMSCatalogServer.java

```diff
@@ -0,0 +1,169 @@
+  }
+
+  /**
+   * Convenience method to start a http server that only serves this servlet.
+   * @param conf the configuration
+   * @param catalog the catalog instance to serve
+   * @return the server instance
+   * @throws Exception if servlet initialization fails
+   */
+  public static Server startServer(Configuration conf, HiveCatalog catalog) throws Exception {
+    int port = MetastoreConf.getIntVar(conf, MetastoreConf.ConfVars.ICEBERG_CATALOG_SERVLET_PORT);
+    if (port < 0) {
+      return null;
+    }
+    final HttpServlet servlet = createServlet(conf, catalog);
+    ServletContextHandler context = new ServletContextHandler(ServletContextHandler.NO_SESSIONS);
+    context.setContextPath("/");
+    ServletHolder servletHolder = new ServletHolder(servlet);
+    servletHolder.setInitParameter("javax.ws.rs.Application", "ServiceListPublic");
+    final String cli = MetastoreConf.getVar(conf, MetastoreConf.ConfVars.ICEBERG_CATALOG_SERVLET_PATH);
+    context.addServlet(servletHolder, "/" + cli + "/*");
+    context.setVirtualHosts(null);
+    context.setGzipHandler(new GzipHandler());
+
+    final Server httpServer = createHttpServer(conf, port);
+    httpServer.setHandler(context);
+    LOG.info("Starting HMS REST Catalog Server with context path:/{}/ on port:{}", cli, port);
+    httpServer.start();
+    return httpServer;
+  }
+
+  private static Server createHttpServer(Configuration conf, int port) throws IOException {
+    final int maxThreads = conf.getInt(JETTY_THREADPOOL_MAX, 256);
+    final int minThreads = conf.getInt(JETTY_THREADPOOL_MIN, 8);
+    final int idleTimeout = conf.getInt(JETTY_THREADPOOL_IDLE, 60_000);
+    final QueuedThreadPool threadPool = new QueuedThreadPool(maxThreads, minThreads, idleTimeout);
+    final Server httpServer = new Server(threadPool);
+    final SslContextFactory sslContextFactory = ServletSecurity.createSslContextFactory(conf);
+    final ServerConnector connector = new ServerConnector(httpServer, sslContextFactory);
+    connector.setPort(port);
+    connector.setReuseAddress(true);
+    httpServer.setConnectors(new Connector[] {connector});
+    for (ConnectionFactory factory : connector.getConnectionFactories()) {
+      if (factory instanceof HttpConnectionFactory) {
+        HttpConnectionFactory httpFactory = (HttpConnectionFactory) factory;
+        HttpConfiguration httpConf = httpFactory.getHttpConfiguration();
+        httpConf.setSendServerVersion(false);
+        httpConf.setSendXPoweredBy(false);
+      }
+    }
+    return httpServer;
+  }
+
+
+  /**
+   * Convenience method to start a http server that only serves this servlet.
+   * <p>This one is looked up through reflection to start from HMS.</p>
+   * @param conf the configuration
+   * @return the server instance
+   * @throws Exception if servlet initialization fails
+   */
+  public static Server startServer(Configuration conf) throws Exception {
+    return startServer(conf, null);
```

Comment:
Why do we need a separate server instance? Could we reuse existing server in Hive? There is a HTTPServer in HiveServer2 already.

Reply:
It has to be a separate instance. In my mind it would be even better if it will be a separate service. But I do not get where is another HTTPServer. It is inside the HMS. It is not the HS2.

Reply:
Jetty is lightweight and strict separation of concerns seemed a simpler idea.

Reply:
I could not find a way to create one Jetty but 3 connectors on 3 different ports that would serve 3 different servlets. The advice I could find stated it is easier to create different instances. I created a helper class to be consistent in how such Jetty instances (1 port, 1 servlet) are created

Reply:
Jetty (like Tomcat) is a servlet container that can host multiple servlets within the same instance. You don’t need a separate Jetty instance for each servlet because Jetty is designed to manage multiple servlets efficiently in a single process. 
Each Jetty instance runs as a separate JVM process, consuming system resources (CPU, memory, ports, threads).
If you start a separate Jetty server for each servlet, you waste resources running multiple JVMs, which can cause unnecessary overhead. 
Using a single Jetty instance allows efficient request routing, shared memory, and optimized resource management.
A better approach is to configure different servlets with different context paths inside a single Jetty instance.

Reply:
Jetty is embedded, is a lightweight component as it does not start a new process per instance, it resides within the HMS JVM. Each of the exposed servlets use different ports today which seems to preclude using on Jetty instance (since there does not seem to be a way with Jetty 9 to have different connectors exclusively serving different servlets).

Reply:
You are right, in embedded mode each instance runs within the same JVM. 
However, each Jetty instance still creates separate thread pools, HTTP connectors, and class loaders. That results in redundant resource allocation and introduces memory and CPU overhead.

Reply:
About the servlet being on different ports, I did not want to assume that the property and rest catalog 'services' would have to share the port, routing/port mapping being often used for security (abundance of caution).
That being said, I was wrong wrt Jetty/ports and was able to let the ServerServletBuilder accommodate all cases; there is only one Jetty instance created for both servlets whether they share the port or not. Their ports are still kept as separate variables so one can specify not to instantiate either servlet (-1).